contentcuration/contentcuration/frontend/shared/data/serverSync.js

```diff
@@ -353,23 +404,71 @@ let intervalTimer;
 export function startSyncing() {
   // Initiate a sync immediately in case any data
   // is left over in the database.
-
+  debouncedSyncChanges();
   const websocketUrl = new URL(
     `/ws/sync_socket/${window.CHANNEL_EDIT_GLOBAL.channel_id}/`,
     window.location.href
   );
   websocketUrl.protocol = window.location.protocol == 'https:' ? 'wss:' : 'ws:';
+
   socket = new WebSocket(websocketUrl);
 
   // Connection opened
   socket.addEventListener('open', () => {
     console.log('Websocket connected');
   });
 
-  debouncedSyncChanges();
+  // Listen for any errors due to which connection may be closed.
+  socket.addEventListener('error', event => {
+    console.log('WebSocket error: ', event);
+  });
+
+  socket.addEventListener('message', e => {
+    const data = JSON.parse(e.data);
+    console.log(data);
+    if (data.task) {
+      Task.setTasks([data.task]);
+    }
+    if (data.response_payload && data.response_payload.allowed) {
+      try {
+        handleAllowed(data.response_payload.allowed);
+      } catch (err) {
+        console.log(err);
+      }
+    }
+    if (data.response_payload && data.response_payload.disallowed) {
+      try {
+        handleDisallowed(data.response_payload.disallowed);
+      } catch (err) {
+        console.log(err);
+      }
+    }
+    if (data.change) {
+      try {
+        handleReturnedChanges([data.change]);
+        handleMaxRevs([data.change], data.change.created_by_id);
```

Comment:
The logic in `handleMaxRevs` uses `userId` when processing user related changes by checking `user_id` on the change object. There is a case where a change object would have `user_id !== created_by_id`, when a user invites another user to edit a channel. That means your code here would be open to causing a regression. The existing code always calls `handleMaxRevs` with user's ID obtained from `Session.getSession()`. Maintaining that is a good idea.

As I mentioned on slack, for handling the errors like you sent me, the `created_by_id` would probably work fine, but isn't a good idea for these other calls to `handleMaxRevs`. So you might as well change all of them.